lib/view_model/migrator.rb

```diff
@@ -75,6 +75,34 @@ def migrate_viewmodel!(_view_name, _version, _view_hash, _references)
   class UpMigrator < Migrator
     private
 
+    def migrate_tree!(node, references:)
+      if node.is_a?(Hash) && node[ViewModel::TYPE_ATTRIBUTE] == ViewModel::ActiveRecord::FUNCTIONAL_UPDATE_TYPE
+        migrate_functional_update!(node, references: references)
+      else
+        super
+      end
+    end
+
+    NESTED_FUPDATE_TYPES = ['append', 'update'].freeze
+
+    # The functional update structure uses `_type` internally with a
+    # context-dependent meaning. Retrospectively this was a poor choice, but we
+    # need to account for it here.
+    def migrate_functional_update!(node, references:)
+      actions = node[ViewModel::ActiveRecord::ACTIONS_ATTRIBUTE]
+      return unless actions
+
+      actions.each do |action|
```

Comment:
Consistency nitpick: here you use a guard for nil actions, below you use `values&.each` for absent values. Is this intentional?

Reply:
No, I changed one and not the other, I'll make it consistent.